tests/waku_lightpush/test_client.nim

```diff
@@ -0,0 +1,322 @@
+
+      # Then the message is received by the server
+      check:
+        publishResponse3.isOk()
+        (pubsubTopic, message3) == (await handlerFuture.waitForResult()).value()
+      echo "1"
+      # When publishing the 1MiB + 63KiB + 911B payload (1113999B)
+      handlerFuture = newPushHandlerFuture()
+      let publishResponse4 = await client.publish(pubsubTopic, message4, serverRemotePeerInfo)
+      echo "2"
+      # Then the message is received by the server
+      check:
+        publishResponse4.isOk()
+        (pubsubTopic, message4) == (await handlerFuture.waitForResult()).value()
+      echo "3"
+      # When publishing the 1MiB + 63KiB + 912B payload (1114000B)
+      handlerFuture = newPushHandlerFuture()
+      echo "31"
+      let publishResponse5 = await client.publish(pubsubTopic, message5, serverRemotePeerInfo)
+      echo "4"
+      # Then the message is not received by the server
+      check:
+        not publishResponse5.isOk()
+        (await handlerFuture.waitForResult()).isErr()
+
+    asyncTest "Invalid Encoding Payload":
+      # Given a payload with an invalid encoding
+      let fakeBuffer = @[byte(42)]
+
+      # When publishing the payload
+      let publishResponse = await server.handleRequest(clientPeerId, fakeBuffer)
+
+      # Then the response is negative
+      check:
+        publishResponse.requestId == ""
+      
+      # And the error is returned
+      let response = publishResponse.response.get()
+      check:
+        response.isSuccess == false 
+        response.info.isSome()
+        scanf(response.info.get(), decodeRpcFailure)
+
+    asyncTest "Handle Error":
+      # Given a lightpush server that fails
+      let 
+        handlerError = "handler-error"
+        handlerFuture2 = newFuture[void]()
+        handler2 = proc(peer: PeerId, pubsubTopic: PubsubTopic, message: WakuMessage): Future[WakuLightPushResult[void]] {.async.} =
+          handlerFuture2.complete()
+          return err(handlerError)
+
+      let
+        serverSwitch2 = newTestSwitch()
+        server2 = await newTestWakuLightpushNode(serverSwitch2, handler2)
+      
+      await serverSwitch2.start()
+
+      let serverRemotePeerInfo2 = serverSwitch2.peerInfo.toRemotePeerInfo()
+
+      # When publishing a payload
+      let publishResponse = await client.publish(pubsubTopic, message, serverRemotePeerInfo2)
+
+      # Then the response is negative
+      check:
+        handlerError == publishResponse.error()
+        (await handlerFuture2.waitForResult()).isOk()
+
+      # Cleanup
+      await serverSwitch2.stop()
+
+  suite "Verification of PushResponse Payload":
+    asyncTest "Positive Responses":
+      # When sending a valid PushRequest
+      let publishResponse = await client.publish(pubsubTopic, message, serverRemotePeerInfo)
+
+      # Then the response is positive
+      check publishResponse.isOk()
+    
+    # TODO: Improve: Add more negative responses variations
```

Comment:
IMO, remove the TODO, add a github issue, or do it.

Reply:
Added as an issue, as I don't know how to add more variations.
Ref: https://github.com/waku-org/nwaku/issues/2330